grafana-plugin/src/models/alertgroup/alertgroup.ts

```diff
@@ -44,7 +44,12 @@ export class AlertGroupStore extends BaseStore {
   incidentsItemsPerPage?: number;
 
   @observable
-  alertsSearchResult: any = {};
+  alertsSearchResult: {
+    prev?: string;
+    next?: string;
+    results?: any;
+    page_size?: number;
+  } = {};
```

Comment:
Correct me if I'm wrong but the only place where we update this observable is this one:
![image](https://github.com/grafana/oncall/assets/12073649/75a7efa0-e644-4d42-b1cc-771cfcd1e687)

So the code above assumes it is nested object like `Record<string, {...}>`. So I would say that either we update this type to reflect reality or we refactor updating this observable and getting its value to not use nested but a flat structure.

If I miss sth here or you want to discuss, pls ping me:)

Reply:
And additionally seems that the only usage puts array of strings as `results`, can we reflect that in type instead of using `any`?